decision-records/005-repository-organisation-for-v5.md

```diff
@@ -0,0 +1,120 @@
+# Decision Record Title
+
+Update our repository organisation to prepare v5
+<!-- The title should be a short present tense imperative phrase, less than 50
+     characters, like a git commit message. -->
+
+## Decision
+
+<!-- An overview of the decision made. -->
+
+To prepare our work for v5, we've agreed to make the following changes to the files in our repository and their location:
+
+1. Move the sources (currently `src` ) and built package files (currently `package`) under the same npm package. We'll keep shipping only the built files to npm for now, excluding the source files from the package (either via `.npmignore` or using the `files` field of `package.json`)
+2. Remove the built package files (currently `package`) from the versioning
+3. Update our release process to make sure we keep checking the changes (possibly manually) between the package to be released and the previous version
+4. Leave `dist` in its current location
+
+This will yield the following structure for our repository:
```

Comment:
Is it worth including the prototype kit config in the tree?

As discussed with the Prototype team on Friday, it needs to remain at the root of the npm package.

Reply:
> As discussed with the Prototype team on Friday, it needs to remain at the root of the npm package.

@36degrees Don't forget if they're using `require()` or `import` with Node.js v12.19.0+ we can do:

```json
"exports": {
  ".": {
    "sass": "./govuk/all.scss",
    "import": "./govuk-esm/all.mjs",
    "require": "./govuk/all.js"
  },
  "./govuk/": "./govuk/",
  "./govuk-esm/": "./govuk-esm/",
  "./*.json": "./dist/*.json"
},
```

Where root package `*.json` imports are mapped to `dist/*.json` exports

Reply:
Sounds sensible as a starting point. If we're not 100% sure on the approach I guess we could just list this as a thing we need to explore under 'Implications'?

Reply:
Yeah course, can move it later

Might get some warnings at some point about building from `../` relative to source (Rollup, TypeScript compiler)

Reply:
Oh wait, to me we'd still have the source to generate the JSON file in `src` as we do now. It's the output that would end at the root of the package and not be versionned for simplicity 😄

Reply:
Added an implication and we can pick the implementation details in https://github.com/alphagov/govuk-frontend/issues/2639 when we update our Prototype Kit configuration.